synapse/handlers/appservice.py

```diff
@@ -158,6 +163,90 @@ async def handle_room_events(events):
             finally:
                 self.is_processing = False
 
+    async def notify_interested_services_ephemeral(
```

Comment:
Do we have any thoughts on how heavy this function might be? It looks like it might be called in the background though?

Can you add a docstring talking a bit more about how / when it runs?

Reply:
@Half-Shot Any idea on the initial question here?

Reply:
I wouldn't expect the function to be too heavy on processing, but it calls some potentially expensive DB statements. I wouldn't expect this too be too horrible though, as sufficiently large homeservers should be able to use an appservice worker to offload the work.

I added some information about when it's called in the docstring. Let me know if you need more info.